lib/widget/triple_state_switch.dart

```diff
@@ -0,0 +1,296 @@
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/framework/extensions.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/widgets.dart';
+import 'package:ensemble/widget/input/form_helper.dart';
+import 'package:ensemble/framework/action.dart' as framework;
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+
+class EnsembleTripleStateSwitch extends StatefulWidget
+    with
+        Invokable,
+        HasController<TripleStateSwitchController,
+            EnsembleTripleStateSwitchState> {
+  static const type = 'TripleStateSwitch';
+  EnsembleTripleStateSwitch({Key? key}) : super(key: key);
+
+  final TripleStateSwitchController _controller = TripleStateSwitchController();
+  @override
+  TripleStateSwitchController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => EnsembleTripleStateSwitchState();
+
+  @override
+  Map<String, Function> getters() {
+    return {'value': () => _controller.value.name};
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'value': (value) => _controller.value =
+          SwitchState.values.from(Utils.getString(value, fallback: 'start')) ??
```

Comment:
You have two "start" values here plus the default "start" - that is 3 different places to change if you decide to change the default state. Potential cause for bug unnecessary.
Use this: 
`_controller.value = SwitchState.values.from(value) ?? _controller.value`

Reply:
Done